### ISSUE TITLE
CORS doesn't work via the proxy

### ISSUE THREAD

**Issue by far-blue** [jwilder/nginx-proxy#167]
I was trying to work with an apache server replying to CORS requests over a REST API and while it works in a native environment it was failing when proxy'd. Having done a quick check with tcpdump, it looks like OPTIONS requests are being converted to GET requests by the proxy, causing things to break.

**Comment by programster** [comment]
I had the same issue, so I tried to remedy it adding the `ALLOW_CROSS_DOMAIN_REQUESTS` environment variable to the website container, and altering the `nginx.tmpl` of the proxy by adding:

```
{{ range $container := $containers }}
        {{ if $container.Env.ALLOW_CROSS_DOMAIN_REQUESTS }}
            include /root/app/cors_support;
        {{ end }}
    {{ end }}
```
within the server block, just before the location block. It's very messy, but I've never used Go before, and I must admit that I don't really understand why a host would have multiple containers? Pretty sure this will not work as expected if $containers holds more than one item.

The /root/app/cors_support is just where I placed [this nginx cors support file](https://gist.github.com/Stanback/7145487), but I had to comment out this line:

```
add_header 'Access-Control-Allow-Origin' "$http_origin";
```
This was because I was somehow always setting the header twice, with * and with the origin of where the request was coming from, which is apparently not allowed.

![selection_257](https://cloud.githubusercontent.com/assets/5709838/8192659/93c262d0-1467-11e5-97ad-068fee9f2f99.png)

I think that in an ideal world, the allowed origins would be set in an environment variable of the web application container, which the reverse proxy would read from and use, much like the `VIRTUAL_HOST`. The application behind it would have to not set the header. I've never tested this though, and only managed to get it hacked together with the restriction provided by the website rather than the proxy.

All progress I have made can be found at [programster/nginx-proxy](*https://github.com/programster/nginx-proxy)
